### PR TITLE
WIP - <SelectableList/> - new component

### PR DESCRIPTION
**Pull request by mykas**
#### A wrapper that makes a list clickable.

```jsx
import React from 'react';
import Thumbnail from 'wix-style-react/Thumbnail';

class SelectableExample extends React.Component {
  _onSelect = (event, item) => {};

  _onDeselect = (event, item) => {};

  render() {
    return (
      <SelectableList
        limit={2}
        onSelect={this._onSelect}
        onDeselect={this._onDeselect}
        trigger="selected"
      >
        <Thumbnail src="small" />
        <Thumbnail src="medium" />
        <Thumbnail src="large" />
      </SelectableList>
    );
  }
}
```

### CODE REVIEW

File: src/SelectableList/SelectableList.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  static displayName = 'SelectableList';
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: mapChildrenToId(props.children),
+      selected: [],
+    };
+  }
+  _change = (e, key) => {
```
Comment (argshook) [inline src/SelectableList/SelectableList.js]: let `e` be `event` cause what if i think `e` stands for `error`

File: src/SelectableList/SelectableList.spec.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import Checkbox from '../Checkbox';
+import { checkboxTestkitFactory } from '../../testkit';
+
+describe('SelectableList', () => {
+  const createDriver = createUniDriverFactory(
+    selectableListPrivateDriverFactory,
+  );
+
+  it('should render', async () => {
```
Comment (argshook) [inline src/SelectableList/SelectableList.spec.js]: no need for this, smoke test should cover it. `should render` is more important in e2e with screenshot

File: src/ColorPicker/ColorPicker.js [hunk 3]
```diff
@@ -66,12 +66,7 @@ export default class ColorPicker extends WixComponent {
   }
 
   render() {
-    const {
```
[the remark this answers is not in the document]
Comment (mykas) [reply src/ColorPicker/ColorPicker.js]: Yes, I will remove these.

File: src/SelectableList/SelectableList.spec.js (hunk 2 above)
Comment (mykas) [reply src/SelectableList/SelectableList.spec.js]: ok

File: src/SelectableList/SelectableList.js (hunk 1 above)
Comment (mykas) [reply src/SelectableList/SelectableList.js]: ok

**Comment by hepiyellow** [comment]
## Where is the `README.DESIGN-SPEC.md` ?

First of all, there has been an effort in core3 to develop a super-generic selection list.
Here is the [PR](https://github.com/wix/wix-ui/pull/802) and the [design-spec](https://github.com/wix/wix-ui/blob/d3fcbdcbfa390cfc3e6544f7e86088730137a10c/packages/wix-ui-core/src/components/selection-list/spec.md)

We don't have to build such a full solution, but let's have a design spec here.
Decide on the scope (what features we want now, and which ones could be added later), and have all use-cases in mind, so we can find an API which is forward compatible.

**Comment by argshook** [comment]
## it does not exist

@hepiyellow thanks for your input. this component is a WIP and part of effort together with Promote team to help them create Shoutout Composer 2.0

we don't do it how core3 did because:

1. we want to move gradually forward, not get into endless discussions to make everyone happy about the API
2. we don't want to solve everything immediately
3. we don't want to hold Promote team
4. we have different requirements
5. we want to make flexible reusable component, not something that knows about specifics (like `typeAheadValue` or something)

**Comment by mykas** [comment]
Dismissed component development. Component is not as useful as just writing your own handlers.  User use case example: https://gist.github.com/mykas/3abc8eff86d1fed50e9d259b108a010e

**Pull request closed without merging**